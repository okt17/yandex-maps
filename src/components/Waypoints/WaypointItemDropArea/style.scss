.app__waypoints__drop-area {
  height: 10px;

  &_active {
    background-color: #ffdb4d;
  }
}
