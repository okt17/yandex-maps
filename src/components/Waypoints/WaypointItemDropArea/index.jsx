import React from 'react';
import PropTypes from 'prop-types';
import './style';

const BCN = 'app__waypoints__drop-area'; // Base Class Name

class WaypointItemDropArea extends React.PureComponent {
  state = {};
  static propTypes = {
    index: PropTypes.number.isRequired,
    moveWaypoint: PropTypes.func.isRequired,
  };
  handleDrop = (event) => {
    event.preventDefault();

    const index = parseInt(event.dataTransfer.getData('waypoint'), 10);

    if (Number.isNaN(index) === false) {
      this.props.moveWaypoint(index, this.props.index);
    }

    if (this.state.active) {
      this.setState({ active: false });
    }
  };
  handleDragEnter = (event) => {
    event.preventDefault();

    if (event.dataTransfer.items[0].type === 'waypoint') {
      this.setState({ active: true });

      return true;
    }

    return false;
  };
  handleDragLeave = (event) => {
    event.preventDefault();

    if (this.state.active) {
      this.setState({ active: false });
    }
  };
  handleDragOver = (event) => {
    event.preventDefault();
  };
  render() {
    const { active } = this.state;

    return (
      <div
        className={`${BCN}${active ? ` ${BCN}_active` : ''}`}
        onDrop={this.handleDrop}
        onDragEnter={this.handleDragEnter}
        onDragLeave={this.handleDragLeave}
        onDragOver={this.handleDragOver}
      />
    );
  }
}

export default WaypointItemDropArea;
