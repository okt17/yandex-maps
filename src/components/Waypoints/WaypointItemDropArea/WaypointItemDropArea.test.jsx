/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import WaypointItemDropArea from '.';

test('WaypointItemDropArea test', () => {
  const component = renderer.create(<WaypointItemDropArea
    index={0}
    moveWaypoint={() => {}}
  />);

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
