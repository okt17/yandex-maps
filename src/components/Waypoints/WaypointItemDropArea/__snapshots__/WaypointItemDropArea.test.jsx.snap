// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WaypointItemDropArea test 1`] = `
<div
  className="app__waypoints__drop-area"
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDrop={[Function]}
/>
`;
